### PR TITLE
[stable/nginx-ingress] Fix ClusterRole creation when using namespaces

### LINKED ISSUE #18420: [stable/nginx-ingress] Enabling controller.scope.enabled will disable ClusterRole creation
**Describe the bug**
Enabling controller.scope.enabled in the values will stop creation of the ClusterRole.
Looking at the following lines:
https://github.com/helm/charts/blob/c0a7ce93fe4114a73415e2963989c021610a5fee/stable/nginx-ingress/templates/clusterrole.yaml#L1-L23
The if statement on L23 will never be hit due to it being within an IF statement on L1 which doesn't trigger if the controller.scope.enabled is enabled.

**Version of Helm and Kubernetes**:
Helm: v2.14.3
Kubernetes: v2.14.3

**Which chart**:
stable/nginx-ingress

**What happened**:
No ClusterRole was created after enabling controller.scope.enabled

**What you expected to happen**:
A valid ClusterRole with specific namespaces to to watch

**How to reproduce it** (as minimally and precisely as possible):
Enable: controller.scope.enabled=true

**Anything else we need to know**:
n/a

### PR DESCRIPTION
**Pull request by Graham42**
#### Is this a new chart
no

#### What this PR does / why we need it:
We want to be able to choose to deploy to a namespace and still opt-in
to ClusterRole creation.

#### Which issue this PR fixes

  - fixes #18420 

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/nginx-ingress/templates/clusterrole.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.rbac.create (not .Values.controller.scope.enabled) -}}
+{{- if and .Values.rbac.create -}}
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/templates/clusterrole.yaml]: you will also need to remove the `and`

**Review by ChiefAlexander — CHANGES_REQUESTED** [review]

File: stable/nginx-ingress/templates/clusterrole.yaml (hunk 1 above)
Comment (Graham42) [reply stable/nginx-ingress/templates/clusterrole.yaml]: Thanks! Fixed now

**Comment by ChiefAlexander** [comment]
/lgtm
Thanks!

**Pull request merged**